cpp/dolfinx/fem/interpolate.h

```diff
@@ -646,4 +650,51 @@ void interpolate(Function<T>& u, const Function<T>& v)
   }
 }
 
+/// Interpolate from an Expression into a compatible Function on the
+/// same mesh
+/// @param[out] u The function to interpolate into
+/// @param[in] expr The expression to be interpolated
```

Comment:
The conditions on the Expression need to be clear and explicit, e.g. the evaluation point locations.